src/Model/GContact.php

```diff
@@ -0,0 +1,2584 @@
+			intval($cid),
+			intval($uid),
+			intval($gcid),
+			intval($zcid)
+		);
+	
+		if (!DBM::is_result($r)) {
+			q(
+				"INSERT INTO `glink` (`cid`, `uid`, `gcid`, `zcid`, `updated`) VALUES (%d, %d, %d, %d, '%s') ",
+				intval($cid),
+				intval($uid),
+				intval($gcid),
+				intval($zcid),
+				dbesc(datetime_convert())
+			);
+		} else {
+			q(
+				"UPDATE `glink` SET `updated` = '%s' WHERE `cid` = %d AND `uid` = %d AND `gcid` = %d AND `zcid` = %d",
+				dbesc(datetime_convert()),
+				intval($cid),
+				intval($uid),
+				intval($gcid),
+				intval($zcid)
+			);
+		}
+	}
+
+	public static function pocoReachable($profile, $server = "", $network = "", $force = false)
+	{
+		if ($server == "") {
+			$server = self::pocoDetectServer($profile);
+		}
+	
+		if ($server == "") {
+			return true;
+		}
+	
+		return self::pocoCheckServer($server, $network, $force);
+	}
+
+	public static function pocoDetectServer($profile)
+	{
+		// Try to detect the server path based upon some known standard paths
+		$server_url = "";
+	
+		if ($server_url == "") {
+			$friendica = preg_replace("=(https?://)(.*)/profile/(.*)=ism", "$1$2", $profile);
+			if ($friendica != $profile) {
+				$server_url = $friendica;
+				$network = NETWORK_DFRN;
+			}
+		}
+	
+		if ($server_url == "") {
+			$diaspora = preg_replace("=(https?://)(.*)/u/(.*)=ism", "$1$2", $profile);
+			if ($diaspora != $profile) {
+				$server_url = $diaspora;
+				$network = NETWORK_DIASPORA;
+			}
+		}
+	
+		if ($server_url == "") {
+			$red = preg_replace("=(https?://)(.*)/channel/(.*)=ism", "$1$2", $profile);
+			if ($red != $profile) {
+				$server_url = $red;
+				$network = NETWORK_DIASPORA;
+			}
+		}
+	
+		// Mastodon
+		if ($server_url == "") {
+			$mastodon = preg_replace("=(https?://)(.*)/users/(.*)=ism", "$1$2", $profile);
+			if ($mastodon != $profile) {
+				$server_url = $mastodon;
+				$network = NETWORK_OSTATUS;
+			}
+		}
+	
+		// Numeric OStatus variant
+		if ($server_url == "") {
+			$ostatus = preg_replace("=(https?://)(.*)/user/(.*)=ism", "$1$2", $profile);
+			if ($ostatus != $profile) {
+				$server_url = $ostatus;
+				$network = NETWORK_OSTATUS;
+			}
+		}
+	
+		// Wild guess
+		if ($server_url == "") {
+			$base = preg_replace("=(https?://)(.*?)/(.*)=ism", "$1$2", $profile);
+			if ($base != $profile) {
+				$server_url = $base;
+				$network = NETWORK_PHANTOM;
+			}
+		}
+	
+		if ($server_url == "") {
+			return "";
+		}
+	
+		$r = q(
+			"SELECT `id` FROM `gserver` WHERE `nurl` = '%s' AND `last_contact` > `last_failure`",
+			dbesc(normalise_link($server_url))
+		);
+
+		if (DBM::is_result($r)) {
+			return $server_url;
+		}
+	
+		// Fetch the host-meta to check if this really is a server
+		$serverret = z_fetch_url($server_url."/.well-known/host-meta");
+		if (!$serverret["success"]) {
+			return "";
+		}
+	
+		return $server_url;
+	}
+
+	public static function poco_alternate_ostatus_url($url)
```

Comment:
Please rename to `pocoAlternateOStatusURL`.

Reply:
I think we can get rid of the "poco" as well. We only used "poco_" as some kind of namespace.